### ISSUE TITLE
 GA4 | Add data layer event for commento comment submission

### ISSUE THREAD

**Issue by danielfmiranda** [mozilla/foundation.mozilla.org#9336]
This is a follow up ticket to #9271 as we had to reach out to commento.io support before starting this work.

The following  is a populated example of a  data layer push that should take place whenever a comment is submitted

Example form: https://foundation.mozilla.org/en/privacynotincluded/threema/

```
window.dataLayer = window.dataLayer || [];
window.dataLayer.push({
	'event' : 'form_submission',
	'form_name' : 'Threema',
	'form_location' : 'foundation.mozilla.org/en/privacynotincluded/threema/',
	'form_type' : 'petition-form',
         'form_id' : '8611'
});

```

**Comment by danielfmiranda** [comment]
Just providing a quick update, I have reached out to commento.io support and have added @cdanfon to the BCC list, however, after ~ 7 days and a follow up there has been no response.

@cdanfon I am wondering if this means you would like me to work on implementing this through the javascript method I mentioned in #9271?

**Comment by cdanfon** [comment]
Hey @danielfmiranda thanks for the update! Let's go with option 1.

**Comment by danielfmiranda** [comment]
Providing an update on this ticket:
- I have tested out my code through the google analytics preview tool using the steps found in ticket #9387, and can confirm that it works when testing using that URL.
- However, when testing on staging, the event does not seem to fire.

I am going to continue to investigate this

**Comment by danielfmiranda** [comment]
One possible problem could be that since these commento elements are being load through commentos JS, our minified JS including the file that adds the event handlers runs before these objects even exist.

For example, here is a screenshot where I set a breakpoint where we do a querySelector for the commento submit button:
<img width="1493" alt="image" src="https://user-images.githubusercontent.com/18314510/191384486-5f7d7a4b-c17f-4396-aff9-848fbdb59da0.png">
Note that the commento box has not loaded in yet.

**Comment by danielfmiranda** [comment]
PR #9414 now provides a fix! :fingers-crossed:

**Comment by danielfmiranda** [comment]
Hi @cdanfon, this ticket is now merged into staging and ready for testing from the torchbox team. 
When testing myself, everything seems to be working as expected:
![image.png](https://images.zenhubusercontent.com/6046b7f1bc36adc9ce95a16a/c31728fd-0f94-4d34-86a4-e5b3f67018bc)

Here are the following test cases that I have checked:
- Logged in, and submitting comment
- Attempt to submit a comment but not logged in
- Log in, then submit a comment

Please let me know if there are any questions, thanks!

**Comment by cdanfon** [comment]
@danielfmiranda great news, this has been tested and seems to be working as expected! Ready to be push to prod
